src/g_all_guis.c

```diff
@@ -219,26 +219,33 @@ void iemgui_verify_snd_ne_rcv(t_iemgui *iemgui)
     }
 }
 
-t_symbol *iemgui_new_dogetname(t_iemgui *iemgui, int indx, t_atom *argv)
+t_symbol *iemgui_new_doget_labelname(t_iemgui *iemgui, int indx, t_atom *argv)
 {
     if (IS_A_SYMBOL(argv, indx))
         return (atom_getsymbolarg(indx, 100000, argv));
     else if (IS_A_FLOAT(argv, indx))
     {
         char str[80];
-        sprintf(str, "%d", (int)atom_getfloatarg(indx, 100000, argv));
+        sprintf(str, "%g", atom_getfloatarg(indx, 100000, argv));
         return (gensym(str));
     }
     else return (gensym("empty"));
 }
 
+t_symbol *iemgui_new_dogetname(t_iemgui *iemgui, int indx, t_atom *argv)
+{
+    if (IS_A_SYMBOL(argv, indx))
+        return (atom_getsymbolarg(indx, 100000, argv));
+    return (gensym("empty"));
+}
+
 void iemgui_new_getnames(t_iemgui *iemgui, int indx, t_atom *argv)
 {
     if (argv)
     {
         iemgui->x_snd = iemgui_new_dogetname(iemgui, indx, argv);
         iemgui->x_rcv = iemgui_new_dogetname(iemgui, indx+1, argv);
-        iemgui->x_lab = iemgui_new_dogetname(iemgui, indx+2, argv);
+        iemgui->x_lab = iemgui_new_doget_labelname(iemgui, indx+2, argv);
```

Comment:
hmm.
introducing a new function to be called exactly once seems to be an overkill.
how about:
- extending `iemgui_new_dogetname` to also return values for `A_FLOAT`
- use `!(IS_A_FLOAT(argv, indx))` to special-case the snd/rcv names

alternatively, just add a `IS_A_FLOAT()` clause to handle the float-case for labels.